Lib/fontTools/ttLib/ttFont.py

```diff
@@ -556,12 +558,22 @@ def getGlyphID(self, glyphName):
 					except (NameError, ValueError):
 						raise KeyError(glyphName)
 
-		glyphID = d[glyphName]
 		if glyphName != glyphOrder[glyphID]:
```

Comment:
Under what circumstances would this condition be true? That is, how can the glyph order list and the reverse map get unsynchronized?

Reply:
Only if client modifies the glyphOrder list manually after it has already been used to build reverse list.

Reply:
Does that need to be supported? In https://github.com/fonttools/fonttools/pull/1654#issuecomment-505541892, @anthrotype wrote:

> Direct usages of glyphOrder attribute outside of get/setGlyphOrder were never supported. I don't think even fontTools library does that internally, even though technically it could. So I am not worried too much if user code is relying on that attribute.

Reply:
> Does that need to be supported? In [#1654 (comment)](https://github.com/fonttools/fonttools/pull/1654#issuecomment-505541892), @anthrotype wrote:
> 
> > Direct usages of glyphOrder attribute outside of get/setGlyphOrder were never supported. I don't think even fontTools library does that internally, even though technically it could. So I am not worried too much if user code is relying on that attribute.

I just wasn't convinced, and I given lack of documentation, I wasn't sold on @anthrotype's assertion and didn't want to possibly break that.

That said, I checked AFDKO and couldn't find any trace of that.

I'll add a commit to address that now. Yay.

In that case, I suggest we make `self.glyphOrder` be immutable.  I'll see if there's such a thing.  Also, we should make glyphOrder a property.

Reply:
> I'll add a commit to address that now. Yay.
> 
> In that case, I suggest we make `self.glyphOrder` be immutable. I'll see if there's such a thing. Also, we should make glyphOrder a property.

Can we do these in a separate PR?

Reply:
> I suggest we make `self.glyphOrder` be immutable. I'll see if there's such a thing

I meant other than `tuple`. But I suppose `tuple` works.

Reply:
@dscorbett @anthrotype actually `GlyphOrder`'s own `fromXML` does this:
```
        def fromXML(self, name, attrs, content, ttFont):
                if not hasattr(self, "glyphOrder"):
                        self.glyphOrder = []
                        ttFont.setGlyphOrder(self.glyphOrder)
                if name == "GlyphID":
                        self.glyphOrder.append(attrs["name"])
```

Fixing up..

Reply:
> Fixing up..

Not sure if possible without changing the XML machinery we have though.